extensions/wikia/AchievementsII/services/AchUserProfileService.class.php

```diff
@@ -214,10 +214,13 @@ public function getChallengesAnnotated() {
 			}
 		}
 
-		global $wgEnableAchievementsForSharing;
+		global $wgEnableAchievementsForSharing, $wgAchievementsEditAddPhotoOnly;
 
 		$challengesAnnotated = array();
 		foreach($challenges as $badge_type_id => $badge_lap) {
+			if(!empty($wgAchievementsEditAddPhotoOnly) && ($badge_type_id != BADGE_EDIT && $badge_type_id != BADGE_PICTURE)) {
```

Comment:
You have pretty much exact same statement in AchAwardingService.class.php. Please consider having this logic in one place.